src/java.desktop/share/classes/java/beans/PropertyEditorSupport.java

```diff
@@ -291,19 +291,19 @@ public void firePropertyChange() {
         PropertyChangeEvent evt = new PropertyChangeEvent(source, null, null, null);
 
         for (int i = 0; i < targets.size(); i++) {
-            PropertyChangeListener target = targets.elementAt(i);
+            PropertyChangeListener target = targets.get(i);
             target.propertyChange(evt);
         }
     }
 
     @SuppressWarnings("unchecked")
-    private <T> java.util.Vector<T> unsafeClone(java.util.Vector<T> v) {
-        return (java.util.Vector<T>)v.clone();
+    private <T> ArrayList<T> unsafeClone(ArrayList<T> v) {
+        return (ArrayList<T>)v.clone();
```

Comment:
I think this can be simplified by using the new list in place of "unsafeClone".

Reply:
Updated